src/decoder/stream.rs

```diff
@@ -67,6 +68,44 @@ pub trait EndianReader: Read {
     }
 }
 
+/// Reader that decompresses DEFLATE streams
+pub struct DeflateReader {
+    buffer: io::Cursor<Vec<u8>>,
+    byte_order: ByteOrder,
+}
+
+impl DeflateReader {
+    pub fn new<R: Read + Seek>(
+        reader: &mut SmartReader<R>,
+    ) -> io::Result<(usize, Self)> {
+        let byte_order = reader.byte_order;
+        let mut compressed = Vec::new();
+        reader.read_to_end(&mut compressed)?;
+        let uncompressed = inflate::decompress_to_vec_zlib(&compressed).map_err(|error| {
```

Comment:
Can this read or decompression subvert the byte limitation from `Limits`?

Reply:
It can, but the only way I can forsee fixing this (and increase performance) is by using streaming decompression. In the case of LZW this is supported naturally, but miniz_oxide's documentation currently states that "As of now this (streaming) is mainly inteded for use to build a higher-level wrapper" (referring to decompress_to_vec()). Ideally, streaming would receive first-class support upstream, but until that happens It could be implemented within image-tiff.

Reply:
We could at least ensure that the compressed representation does not exceed it. [The implementation does not appear to be magic](https://docs.rs/miniz_oxide/0.3.2/src/miniz_oxide/inflate/mod.rs.html#73) so maybe it's possible to upstream a wrapper or have our own temporarily.

Reply:
Agreed, I suppose it would be prompt to open a separate issue until the temporary custom solution is written or changes are made upstream to miniz_oxide? For now, ensuring the compressed representation is under the size restriction is a good idea and I'll go ahead and implement it.